external/sql/storm-sql-runtime/src/jvm/org/apache/storm/sql/runtime/trident/operations/CountBy.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.sql.runtime.trident.operations;
+
+import org.apache.storm.sql.runtime.trident.TridentUtils;
+import org.apache.storm.trident.operation.CombinerAggregator;
+import org.apache.storm.trident.tuple.TridentTuple;
+
+
+public class CountBy implements CombinerAggregator<Long> {
+
+    private final String inputFieldName;
+
+    public CountBy(String inputFieldName) {
+        this.inputFieldName = inputFieldName;
+    }
+
+    @Override
+    public Long init(TridentTuple tuple) {
+        if (TridentUtils.valueFromTuple(tuple, inputFieldName) == null) {
+            return 0L;
```

Comment:
Is it supposed to count only the tuples with non-null values for inputFieldName ? If so would it return the right count since combine cant check this?